src/native/external/llvm-libunwind/src/DwarfParser.hpp

```diff
@@ -269,7 +269,7 @@ bool CFI_Parser<A>::findFDE(A &addressSpace, pint_t pc, pint_t ehSectionStart,
           pint_t pcRange = addressSpace.getEncodedP(
               p, nextCFI, cieInfo->pointerEncoding & 0x0F);
           // Test if pc is within the function this FDE covers.
-          if ((pcStart < pc) && (pc <= pcStart + pcRange)) {
+          if ((pcStart <= pc) && (pc < pcStart + pcRange)) {
```

Comment:
Note: This should be upstreamed.

With pure DWARF CFI unwinding it was mostly harmless bug. Unless the offset was in epilog or prolog it would return the correct thing. If it was the very first instruction then it would either not find the method, or find the previous one. That could result in silent errors but usually not a crash.

However, with combined compact and DWARF unwinding there's often a case where the compact unwinding table contains an entry that says to use DWARF (with hint offset into the DWARF table). It would correctly process the compact unwinding entry but fail to find the DWARF one. That resulted in internal unwinder state where it provided some information but noticeably the LSDA pointer was `NULL`. That would get dereferenced and crash the application.

Reply:
Please also pull this change in a separate commit and drop a line (`Apply {commit-sha}`) in `src/native/external/llvm-libunwind-version.txt`. We are tracking these changes for upstream in groups: https://github.com/dotnet/runtime/issues/72655. (your standalone commit will outlive when PR is squashed+merged and branch is deleted)